source/Coop.Core/Server/Services/MobileParties/Handlers/ServerSettlementExitEnterHandler.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Common.Logging;
+using Common.Messaging;
+using Common.Network;
+using Coop.Core.Client.Services.MobileParties.Messages;
+using GameInterface.Services.MobileParties.Messages;
+using LiteNetLib;
+using Serilog;
+
+namespace Coop.Core.Server.Services.MobileParties.Handlers
+{
+    /// <summary>
+    /// Handles changes to parties for settlement entry and exit.
+    /// </summary>
+    public class ServerSettlementExitEnterHandler : IHandler
+    {
+        private readonly IMessageBroker messageBroker;
+        private readonly INetwork network;
+        private readonly ILogger Logger = LogManager.GetLogger<ServerSettlementExitEnterHandler>();
+
+        public ServerSettlementExitEnterHandler(IMessageBroker messageBroker, INetwork network)
+        {
+            this.messageBroker = messageBroker;
+            this.network = network;
+            messageBroker.Subscribe<NetworkSettlementEnterRequest>(Handle);
+        }
+
+        public void Dispose()
+        {
+            messageBroker.Unsubscribe<NetworkSettlementEnterRequest>(Handle);
+        }
+
+        private void Handle(MessagePayload<NetworkSettlementEnterRequest> obj)
+        {
+            NetworkPartyEnteredSettlement partyEnteredSettlement = new NetworkPartyEnteredSettlement(obj.What.SettlementId, obj.What.PartyId);
+
+            network.SendAllBut(obj.Who as NetPeer, partyEnteredSettlement);
```

Comment:
This should also be sent to the party entering the settlement, if the party enters the settlement without the servers permission, this isn't working as intended